libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+    options = {
+      throwIfNotExists: false,
+      ...options
+    };
+
+    if (options.throwIfNotExists || FileSystem.exists(filePath)) {
+      fsx.unlinkSync(filePath);
+    }
+  }
+
+  // ===============
+  // LINK OPERATIONS
+  // ===============
+
+  /**
+   * Gets the statistics of a file, folder, or link. Does NOT follow the link to its target.
+   * Behind the scenes it uses `fs.lstatSync()`.
+   * @param path - The absolute or relative path to the file, folder, or link.
+   */
+  public static getLinkStatistics(path: string): fs.Stats {
+    return fsx.lstatSync(path);
+  }
+
+  /**
+   * Creates a symbolic link to a folder (on Windows this is a "directory junction").
+   * Behind the scenes it uses `fs.symlinkSync()`.
+   * @param linkSource - The absolute or relative path to the destination where the link should be created.
+   * @param linkTarget - The absolute or relative path to the target of the link.
+   */
+  public static createSymbolicLinkToFolder(linkSource: string, linkTarget: string): void {
+    // For directories, we use a Windows "junction".  On Unix, this produces a regular symlink.
+    fsx.symlinkSync(linkTarget, linkSource, 'junction');
+  }
+
+  /**
+   * Creates a symbolic link to a file (on Windows this requires elevated permissions).
+   * Behind the scenes it uses `fs.symlinkSync()`.
+   * @param linkSource - The absolute or relative path to the destination where the link should be created.
+   * @param linkTarget - The absolute or relative path to the target of the link.
+   */
+  public static createSymbolicLinkToFile(linkSource: string, linkTarget: string): void {
+    fsx.symlinkSync(linkSource, linkTarget, 'file');
+  }
+
+  /**
+   * Creates a hard link to a file.
+   * Behind the scenes it uses `fs.linkSync()`.
+   * @param linkSource - The absolute or relative path to the destination where the link should be created.
+   * @param linkTarget - The absolute or relative path to the target of the link.
+   */
+  public static createHardLinkToFile(linkSource: string, linkTarget: string): void {
+    fsx.linkSync(linkSource, linkTarget);
+  }
+
+  /**
+   * Follows a link to its destination and returns the absolute path to the final target of the link.
+   * Behind the scenes it uses `fs.realpathSync()`.
+   * @param linkPath - The path to the link.
+   */
+  public static followLink(linkPath: string): string {
+    return fsx.realpathSync(linkPath);
+  }
+
+  /**
+   * A helper function that converts line endings on a string.
+   * @param text - The text to be normalized.
+   * @param lineEndings - The style of line endings to use.
+   */
+  private static _convertLineEndings(text: string, lineEndings: NewlineConversion | undefined): string {
+    if (lineEndings === NewlineConversion.CrLf) {
+      return Text.convertToCrLf(text);
+    } else if (lineEndings === NewlineConversion.Lf) {
+      return Text.convertToLf(text);
+    }
+    return text;
+  }
```

Comment:
Doesn't this logic belong in the Text API?

Text.convertNewlines()?

Use "newlineKind" instead of "lineEndings"